DQMOffline/Trigger/plugins/TopMonitor.cc

```diff
@@ -0,0 +1,1024 @@
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  //Suvankar
+  edm::Handle<reco::VertexCollection> primaryVertices;
+  iEvent.getByToken(vtxToken_, primaryVertices);
+  //Primary Vertex selection
+  const reco::Vertex* pv = nullptr;
+  for(auto const& v: *primaryVertices) {
+    if ( !vtxSelection_( v ) )      continue;
+    pv = &v;
+    break;
+  }
+  if(usePVcuts_ && pv == nullptr)      return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  if (!metHandle.isValid()){
+      edm::LogWarning("TopMonitor") << "MET handle not valid \n";
+      return;
+  }
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+  
+  float met = pfmet.pt();
+  float phi = pfmet.phi();
+
+  edm::Handle<edm::View<reco::GsfElectron> > eleHandle;
+  iEvent.getByToken( eleToken_, eleHandle );
+  if (!eleHandle.isValid()){
+      edm::LogWarning("TopMonitor") << "Electron handle not valid \n";
+      return;
+  }
+  std::vector<reco::GsfElectron> electrons;
+  if ( eleHandle->size() < nelectrons_ ) return;
+  for ( auto const & e : *eleHandle ) {
+    if ( eleSelection_( e ) ) electrons.push_back(e);
+    //Suvankar
+    if ( usePVcuts_ && 
+         (std::fabs(e.gsfTrack()->dxy(pv->position())) >= lepPVcuts_.dxy || std::fabs(e.gsfTrack()->dz(pv->position())) >= lepPVcuts_.dz) ) continue;
+  }
+  if ( electrons.size() < nelectrons_ ) return;
+  
+  edm::Handle<reco::MuonCollection> muoHandle;
+  iEvent.getByToken( muoToken_, muoHandle );
+  if (!muoHandle.isValid()){
+      edm::LogWarning("TopMonitor") << "Muon handle not valid \n";
+      return;
+  }
+  if ( muoHandle->size() < nmuons_ ) return;
+  std::vector<reco::Muon> muons;
+  for ( auto const & m : *muoHandle ) {
+    if ( muoSelection_( m ) ) muons.push_back(m);
+    //Suvankar
+    if ( usePVcuts_ && 
+         (std::fabs(m.muonBestTrack()->dxy(pv->position())) >= lepPVcuts_.dxy || std::fabs(m.muonBestTrack()->dz(pv->position())) >= lepPVcuts_.dz) ) continue;
+  }
+  if ( muons.size() < nmuons_ ) return;
+
+  double eventHT = 0.;
+
+  edm::Handle<reco::PFJetCollection> jetHandle;
+  iEvent.getByToken( jetToken_, jetHandle );
+  if (!jetHandle.isValid()){
+      edm::LogWarning("TopMonitor") << "Jet handle not valid \n";
+      return;
+  }
+  std::vector<reco::PFJet> jets;
+  if ( jetHandle->size() < njets_ ) return;
+  for ( auto const & j : *jetHandle ) {
+      if ( HTdefinition_ ( j ) ){
+          eventHT += j.pt();
+      }
+      if ( jetSelection_( j ) ){
+          bool isJetOverlappedWithLepton = false;
+          TLorentzVector jp4; jp4.SetPtEtaPhiE(j.pt(),j.eta(),j.phi(),j.energy());
+          if(nmuons_>0){
+              for (auto const m : muons){
+                  TLorentzVector mp4; mp4.SetPtEtaPhiE(m.pt(),m.eta(),m.phi(),m.energy());
+                  if (mp4.DeltaR(jp4)<leptJetDeltaRmin_){
```

Comment:
use DataFormats/Math/include/deltaR and avoid the either conversion to TLorentzVector